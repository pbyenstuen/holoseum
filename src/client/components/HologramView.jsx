import React from 'react';
import { useParams } from "react-router";
import { BlockReserveLoading } from "react-loadingg";
import ErrorView from "./ErrorView";
import useLoader from "./useLoader";

const HologramView = ({ api }) => {
    const { name } = useParams();
    const { loading, error, reload } = useLoader(async () => await api.holo.getHologram(name), [name]);

    if (loading) {
        return <BlockReserveLoading />;
    }

    if (error) {
        return <ErrorView error={error} reload={reload} />;
    }

    return (
        <div id="video-container">
            <video width="100%" height="100%" controls loop autoPlay muted >
                <source src={`/api/holograms/${name}`} type="video/mp4"></source>
            </video>
        </div>
    )
}

export default HologramView;
