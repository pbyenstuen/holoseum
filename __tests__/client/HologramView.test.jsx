import React from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";
import { MemoryRouter } from "react-router";
import HologramView from "../../src/client/components/HologramView";

let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

const mountComponent = async (component) => {
    await act(async () => {
        await ReactDOM.render(<MemoryRouter>{component}</MemoryRouter>, container);
    });
}

describe("HologramView", () => {
    it("can show error", async () => {
        const api = {
            holo: {
                getHologram: () => {
                    throw new Error("Error");
                }
            }
        }

        await mountComponent(<HologramView api={api} />);

        expect(container.innerHTML).toMatchSnapshot();
        expect(container.querySelector("p").textContent).toEqual("Noe gikk galt: Error: Error");
    });
});