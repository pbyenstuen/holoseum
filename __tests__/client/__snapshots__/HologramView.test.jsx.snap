// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HologramView can show error 1`] = `"<div id=\\"error-container\\" class=\\"center-full-screen\\"><h1>Error</h1><p>Noe gikk galt: Error: Error</p><button>PRÃ˜V IGJEN</button></div>"`;
